--- util/mk-mkd	2020-10-16 07:25:25
+++ util/mk-mkd-patched	2023-07-20 09:33:35
@@ -166,7 +166,7 @@
 	        ACC_CMAKE_ARGS=()
 	    fi
 
-	    cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} "${ACC_CMAKE_ARGS[@]}" .. || exit 1
+	    cmake ${CMAKE_ARGS} -DCMAKE_BUILD_TYPE=${BUILD_TYPE} "${ACC_CMAKE_ARGS[@]}" .. || exit 1
 	    ${GMAKE} -j ${ACC_SET_GMAKE_JOBS} ${ACC_EXE_NAME}
 	    [ $? -eq 0 ] && func_display_build_time
 	fi
